streams/src/main/java/org/apache/kafka/streams/state/internals/LeftOrRightValue.java

```diff
@@ -39,45 +39,6 @@ private LeftOrRightValue(final V1 leftValue, final V2 rightValue) {
         this.rightValue = rightValue;
     }
 
-    /**
-     * Create a new {@link LeftOrRightValue} instance with the V1 value as {@code leftValue} and
-     * V2 value as null.
-     *
-     * @param leftValue the left V1 value
-     * @param <V1>      the type of the value
-     * @return a new {@link LeftOrRightValue} instance
-     */
-    public static <V1, V2> LeftOrRightValue<V1, V2> makeLeftValue(final V1 leftValue) {
```

Comment:
I think you should keep the `makeLeftValue` and `makeRightValue` functions as entrypoints to this class. Only the the impossible-to-type `make` function should be removed.